src/parser_TEST.cc

```diff
@@ -353,6 +353,31 @@ TEST(Parser, PlacementFrameMissingPose)
   EXPECT_EQ(sdf::ErrorCode::MODEL_PLACEMENT_FRAME_INVALID, errors[0].Code());
 }
 
+/////////////////////////////////////////////////
+// Delimiter '::' in name should error in SDFormat 1.8 but not in 1.7
+TEST(Parser, DoubleColonNameAttrError)
+{
+  sdf::SDFPtr sdf = InitSDF();
+  std::ostringstream stream;
+  stream << "<?xml version=\"1.0\"?>"
+         << "<sdf version='1.8'>"
+         << "  <model name='test'>"
+         << "    <link name='A::B'/>"
+         << "  </model>"
+         << "</sdf>";
+
+  EXPECT_FALSE(sdf::readString(stream.str(), sdf));
```

Comment:
nit: you could pass an `sdf::Errors` object to `sdf::readString` and check its contents

Reply:
ac6ec01a8bfde7cc3e36fdac8c3f3d6b6ec4ac52

Reply:
looks great!